src/boot/efi/devicetree.c

```diff
@@ -106,6 +106,127 @@ EFI_STATUS devicetree_install(struct devicetree_state *state, EFI_FILE *root_dir
                         MAKE_GUID_PTR(EFI_DTB_TABLE), PHYSICAL_ADDRESS_TO_POINTER(state->addr));
 }
 
+static const char* devicetree_get_compatible(const void *dtb) {
+        if (!IS_ALIGNED64(dtb))
```

Comment:
btw, the 64bit thing where does that come from?

maybe instead: `if ((uintptr_t) dtb % alignof(FdtHeader) != 0)`, just like below?